Leaderboard: Hit the API

BenjaminSemah opened the pull request:
# Leaderboard - Hit the API
## In this branch, I have done the following;
- Create a new game following guidelines from the Leaderboard API documentation,
- Implemented the "Refresh" button (to receive data from the API and parse the JSON),
- Implemented the form "Submit" button (to send user scores data to the API),
- Used `async` and `await` JavaScript features to consume the API.

BenjaminSemah committed:
Add lead-api.js file to modules

BenjaminSemah committed:
Implement refresh button

The refresh button receives scores data from the API and parse the JSON

BenjaminSemah committed:
Implement the form submit button

BenjaminSemah committed:
Fix errors spotted by linters

BenjaminSemah committed:
Update Readme to name of the app

omar-labana commented on src/modules/lead-api.js:
```diff
@@ -0,0 +1,40 @@
+
+const requestURL = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/1FO2b9q2LcMXOmf0R1EU/scores/';
+
+export async function getScores() {
+  const response = await fetch(requestURL);
+  const scoresArr = await response.json();
+  const devScores = scoresArr.result;
+
+  let scoreHTML = '';
+  devScores.forEach((player) => {
+    scoreHTML
+    += `
+    <li class="single-score">
+      <p class="name">${player.user}</p>
+      <p class="score">${player.score}</p>
+    </li>
+    `;
+  });
+  scoreBoard.innerHTML = scoreHTML;
+}
+
+export async function addScores() {
```
- [x] Kindly make sure to use arrow functions instead of the function keyword(es5 is evil).

BenjaminSemah replied:
Resolved ✔️

omar-labana commented on src/modules/lead-api.js:
```diff
@@ -0,0 +1,40 @@
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(
+      {
+        user: userInput.value.trim(),
+        score: scoreInput.value.trim(),
+      },
+    ),
+  });
```
- [x] Kindly, add some validation to make sure that the score value to send to the API is a number.

BenjaminSemah replied:
Resolved ✔️

omar-labana reviewed (changes requested):
Hi @BenjaminSemah, 

Excellent work so far! 👌
You have implemented all the required features correctly and without any errors. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

BenjaminSemah committed:
Add validation to ensure user score is a number

BenjaminSemah committed:
Change functions to ES6 syntax

BenjaminSemah committed:
Fix errors spotted by linters

BenjaminSemah commented:
### All Required Changes has been Implemented.
#### I have implemented the following changes pointed out by first Code Reviewer;
- Changed functions to ES6 using arrow function syntax.
- Added validation to ensure user score input is a number.

mikethreels reviewed (approved):
Hi Benjamin, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
